src/oop_ext/foundation/weak_ref.py

```diff
@@ -18,32 +38,33 @@ class WeakList:
         https://github.com/apieum/weakreflist
     """
 
-    def __init__(self, initlist=None):
-        self.data = []
+    def __init__(self, initlist: Optional[Iterable[T]] = None):
+        self.data: List[SomeWeakRef] = []
 
         if initlist is not None:
             for x in initlist:
                 self.append(x)
 
     @Implements(list.append)
-    def append(self, item):
+    def append(self, item: T) -> None:
```

Comment:
Running mypy with strict returns an error here =/

```
weak_ref.py:49:5: error: Function is untyped after decorator transformation
```

And running:
```
reveal_type( WeakList[_Stub]().append)
```

`note: Revealed type is 'Any'`
Possible is because `decorator.implements` accepts a `Callable` only

Reply:
That's great to know, thanks!

However I couldn't get it to work for the cases here (method decorators, as opposed to function decorators that are shown in the docs).

For example, changing `Override` to:

```python
F = TypeVar('F', bound=Callable[[...], Any])

def Override(method: F) -> Callable[[F], F]:
    ...
```

Which is used like this:

```python
class PriorityCallback(Callback):
    
    @Override(Callback._GetInfo)
    def _GetInfo(self, func):
        ...
```

I get:

```
src\oop_ext\foundation\callback\_priority_callback.py:14: error: Argument 1 has incompatible type "Callable[[PriorityCallback, Any], Any]"; expected "Callable[[Callback, Any], Any]"  [arg-type]
```

Note that the problem is the type of `self`: `Callback` vs `PriorityCallback`.

I looked around the docs and couldn't really find a solution, unfortunately.

Any suggestions? Feel free to clone the repo and play around yourself too.

Reply:
@nicoddemus 

I could solve this in a toy example using two `TypeVar`'s a and one `cast`, perhaps it's the way to solve this problem?

```python
from typing import Callable, Any, TypeVar, cast

F = TypeVar("F", bound=Callable)
G = TypeVar("G", bound=Callable)


def Override(other_method: F) -> Callable[..., F]:
    def _decorate(method: G) -> F:
        return cast(F, method)
    return _decorate


class Base:
    def Method(self):
        ...


class Derived(Base):

    @Override(Base.Method)
    def Method(self):
        ...
```

Reply:
Thanks for the discussion/suggestions folks, managed to get things working in 62d4034. 👍